src/errors.rs

```diff
@@ -1,6 +1,26 @@
 use failure::{Backtrace, Context, Fail};
 use std::fmt::{self, Debug, Display};
 
+/// Kinds of errors which might happen during file upload
+#[derive(Debug, Fail)]
+pub enum FileUploadErrorKind {
+    /// This error will occur when file overriding is off and file with same name already exists
+    #[fail(display = "File with this name already exists")]
+    FileExist,
+    /// This error will occur when server will fail to preccess http header during file upload
```

Comment:
`This error will occur when the server fails to process the HTTP header during file upload`